File: CodeWars/5_kyu/Moving_Zeros_To_The_End/C/main.c

#include <stdio.h>
#include "test.h"

int main() {
    
    int arr[] = {1, 0, 1, 2, 0, 1, 3};
    int length = 7;

    printf("Before : ");
    for (int i = 0; i < length; ++i) {
        printf("%d ", *(arr+i));
    }
    printf("\n");

    move_zeros(length, arr);

    printf("After : ");
    for (int i = 0; i < length; ++i) {
        printf("%d ", *(arr+i));
    }
    printf("\n");

    return 0;
}

File: CodeWars/5_kyu/Moving_Zeros_To_The_End/Python/main.py


def move_zeros(lst):
    for i in range(len(lst)):
        if lst[i] == 0:
            for j in range(i+1, len(lst)):
                if lst[j] != 0:
                    lst[i], lst[j] = lst[j], lst[i]
                    break

    return lst
